### ISSUE TITLE
[Bug]: Failing Integration Tests

### ISSUE THREAD

**Issue by yardasol** [arfc/saltproc#72]
**Expected behavior**
The two integration test cases -- `const_repr`, and `no_repro` -- under `saltproc/tests/integration_tests` should both pass

**Describe the bug**
`const_repr`
Test fails due to input error
```
Input error in parameter "mat" on line 3 in file "/home/ooblack/projects/saltproc/data/saltproc_mat":

Nuclide 47310.09c not found in data libraries
```
Note that the specified nuclide has Z=47. This is Ag (silver). No known isotope of silver with N=310 exists.

`no_repro`
Test fails due to pointer error
```
Fatal error in function GetText:

Invalid pointer: pointer less than valid pointer or grader than the ascii data size

Simulation aborted.
```

**To Reproduce**
1. If you've installed SaltProc in a conda or other virtual environment, activate that environment
2. `cd saltproc`
3. `pytest saltproc/tests/integration_tests`

**Screenshots**
In lieu of screenshots, below are gists to the two error stacks.
[const_repr](https://gist.github.com/yardasol/1ae195a9b7d4243a49a71abb9f6c6b8f)
[no_repro](https://gist.github.com/yardasol/349388949b0b86609bf5946462b3b16f)

**System Specifications:**
 - Version:
   - SaltProc 0.3.dev0 (commit 692f43f)
   - Serpent 2.1.32
   - JEFF 3.1.2 (w/ fission yield and decay data from JEFF 3.3) 
 - Platform: Ubuntu 21.10
 - Subsystem: 
   - Python 3.9.7
   - Conda 4.10.3

**How can this issue be closed?**
This issue can be closed when the integration tests are passing.

**Comment by yardasol** [comment]
## Updates
### `no_repro`
I recompiled Serpent in debug mode and ran the serpent input for the `no_repro` integration test in gdb:
```
INPUT=$HOME/projects/saltproc/saltproc/tests/integration_tests/no_repro/int_test
gdb --args sss2-debug -omp 1 $INPUT
```
[This](https://gist.github.com/yardasol/c75743ed5cb13a02f586f753fa6638e9) is the output I get doing some thread tracing.
The TL;DR is that something is going wrong when serpent is parsing the geometry file provided (`tests/test_geo.inp`).

### `const_repr`
I did a similar process for the `const_repr` case:
```
INPUT=$HOME/projects/saltproc/data/saltproc_serpent
gdb --args ss2-debug -omp 1 $INPUT
```
However, now that we are using debug mode, I get the following error message:
```
Fatal error in function ReadFissionYields:

Value 8.019800E+04 of parameter "ZA" above upper limit 8.000000E+04 

NOTE: This value check was performed because the code was compiled in the
      debug mode and it may or may not be an indication of an actual problem.
      The debugger mode can be switched off by recompiling the source code
      without the -DDEBUG option (see Makefile).
```
My best guess is that this check is trying to make sure we aren't using any fission yields that are unrealistic/unlikely, but my expertise with that is limited so I can't say for sure. @munkm any ideas?

Going back to the issue of the `47310.09c` key, this is present in line 683 in `data/saltproc_mat`, which is generated when we run the integration test. This value is sandwiches within the `471xxx` section, so I believe that the `3` is a typo and should be a `1` instead. Tracking down where these keys are being stored is the next step.

**Comment by yardasol** [comment]
## Update

I set up the `jeff312_all_temps` library from our group box folder on my machine and ran the integration tests.

### `no_repr`
I get the same error as before.

### `cont_repr`
I get the same error of missing materials that are in actuality due to typos in the `xsdata` file. After fixing the typos (see [this](https://github.com/arfc/saltproc/issues/72#issuecomment-951429232) comment and the one immediately below it), I get the following error:

```
Fatal error in function ReadACEFile:

Unable to find isotope 47110.09c in file /home/ooblack/projects/cross-section-libraries/jeff312_all_temps/ACEs_900K/Ag110M.ACE

Simulation aborted.

```

This file does exist at the specified path on my computer, so I don't know why serpent says it can't find it.

I ran this test case within a `gdb` session, set a breakpoint at `readacefile.c:251`, and executed the code. Perfoming a backtrace reveals that one of the variables within this function has the value `HZ1 = "47310.09c...` (where `...` represents many more characters that don't need to be posted here). I checked the `ACEs_900K/Ag110M.ACE` file and confirmed that this character set exists there, at the very top of the file:
```
 47310.09c  108.962000  7.7556E-08   02/29/12
```
I suspect that the rest of the `ACE` files whose corresponding entry in `xsdata` had a typo may also be effected by this.
@gwenchee note that the `jeff312_all_temps` archive on the box is bugged!!

This does _not_ appear to be an issue for the `ACE` files I downloaded from OECD.

**Comment by yardasol** [comment]
## Update

### `const_repr`
For completeness: 
- [Here](https://gist.github.com/yardasol/ad526bb98434a1a9b2bbc0825f2c5a5e) is the gist of running the `const_repr` case with the `jeff312_all_temps` library in debug mode. The TL;DR is that we get the [same error message as before](https://github.com/arfc/saltproc/issues/72#issuecomment-951429232).
- [Here](https://gist.github.com/yardasol/fba19ee3dcddc5540f796c3d64dad774) is the gist of the running the `const_repr` case with the `jeff312_all_temps` library in normal mode with debug headers in `gdb` 

### `const_repr`
I ran this case with `jeff312_all_temps` in `gdb`. When I hit the segfault, I backtraced. [Here](https://gist.github.com/yardasol/6043f3735bf1c3d50535d85a53b8b6e8) is the output of that. My guess is that the code is trying to access an index in the array `RDB` that does not exist. I'm not sure why it would be passing a bad index to the array in the first place. I'll be making a serpent forum account and creating a few posts there to see if I can get any help debugging this.

**Comment by yardasol** [comment]
## Update
### `const_repr`
I made [this](https://ttuki.vtt.fi/serpent/viewtopic.php?f=11&t=3933) post on the serpent forum asking, and shortly thereafter received a reply from Ana Jambrina pointing me to [this](https://ttuki.vtt.fi/serpent/viewtopic.php?f=25&t=3682&start=20#p12675) patch. I applied the patch to the source code, recompiled, and ran the `cons_repr` test case with the `jeff312_all_temps` library, and it passed! (hooray!).

My guess is that these integration tests were written using a previous version of Serpent2 (2.1.30 or 2.1.31 I believe). Updates to Serpent2 included in 2.1.32 may have broken them, but with the aformentioned patch, the `const_repr` case is now working!

**Comment by yardasol** [comment]
All the mysteries of the failing integration tests have been more or less resolved, so I'm closing this.
